lib/core.js

```diff
@@ -547,20 +549,22 @@ type IteratorResult<Yield,Return> =
   | { done: true, value?: Return }
   | { done: false, value: Yield };
 
-interface $Iterator<+Yield,+Return,-Next> {
-    @@iterator(): $Iterator<Yield,Return,Next>;
-    next(value?: Next): IteratorResult<Yield,Return>;
-}
-type Iterator<+T> = $Iterator<T,void,void>;
-
 interface $Iterable<+Yield,+Return,-Next> {
     @@iterator(): $Iterator<Yield,Return,Next>;
 }
 type Iterable<+T> = $Iterable<T,void,void>;
 
-interface Generator<+Yield,+Return,-Next> {
+interface $Iterator<+Yield,+Return,-Next> extends $Iterable<Yield,Return,Next> {
     @@iterator(): $Iterator<Yield,Return,Next>;
```

Comment:
I wish Flow didn't require an `@@iterator` method on Iterators, even though they typically have one that returns `this` by convention. (But I'm not sure how to use this without hacks in any case: #1163.)

Reply:
#6075 has some relevant discussion